src/commands/code.ts

```diff
@@ -159,21 +159,37 @@ export default class CodeCommand extends SlashCommand {
     if (`${lines[actualStart - 1]}`.trim().length <= 0) actualStart++;
     if (`${lines[actualEnd - 1]}`.trim().length <= 0) actualEnd--;
 
-    const lineSelection = lines.slice(actualStart - 1, actualEnd);
-
     let commentOpen = false;
-    for (const [index, line] of lineSelection.entries()) {
-      if (!commentOpen) {
-        lineSelection[index] = line.replace(/^( {2,}) \*/gm, '$1/*');
-        // to ensure comments are always opened if line numbers are enabled
-        if (!shouldHaveLineNumbers) commentOpen = true;
-        amendNotes.add('A comment block was altered for formatting purposes.');
+
+    for (let head = actualStart - 2; head >= 0; head--) {
+      // Comment was opened before the initial head of the selection
+      if (lines[head].indexOf('*/')) {
+        commentOpen = true;
+        break;
       }
+    }
+
+    const lineSelection = lines.slice(actualStart - 1, actualEnd);
 
+    for (const [index, line] of lineSelection.entries()) {
       if (line.indexOf('/*') >= 0) commentOpen = true;
-      if (line.indexOf('*/') >= 0) commentOpen = false;
+      // if (line.indexOf('*/') >= 0) commentOpen = false;
+
+      if (!(commentOpen || shouldHaveLineNumbers)) continue;
+      commentOpen = false;
+
+      const processedLine = line.replace(/^( {2,}) \*/gm, '$1/*');
+
+      if (processedLine === lineSelection[index]) continue;
+
+      lineSelection[index] = processedLine;
+      amendNotes.add('A comment block was altered for formatting purposes.');
     }
 
+    // if (commentOpen) {
+    //   amendNotes.add('A comment block remains open.');
+    // }
```

Comment:
The use of `commentOpen` here is **incorrect**, but leaves the opportunity for another note to be added when it searches the sliced buffer for an open comment (which could also be done if L176 is allowed to continue, and then use the above selection once it is done searching).

Reply:
Also has the certainty to fail if run with `line_numbers` as true for the current handling.